packages/pirania/files/usr/bin/captive-portal

```diff
@@ -14,14 +22,18 @@ clean_tables () {
 			ipaddr=ip6addr
 		fi
 
-		### Cleanup
-		for interface in br-lan anygw; do
-			$iptables -t mangle -D PREROUTING -i $interface -j pirania
-		done
+		if [ $catch_wifi_mode_ifaces = "yes" ] ; then
+			$iptables -t mangle -D PREROUTING -m mark --mark 0x9124714 -j pirania
```

Comment:
i'd unconditionally run this line (without "if", which goes away anyway if you drop the bools in config), and append `2>/dev/null`. if the rule doesn't exist, it will simply fail silently.

Reply:
There are actually no management of silent failures when deleting the pirania chain or rules that are not present. I'll leave this for another PR. Because I think it deserves a bit more of attention, and maybe adding useful logging.